crates/router/src/core/payments/operations/payment_capture.rs

```diff
@@ -225,3 +259,58 @@ impl<F: Send + Clone> ValidateRequest<F, api::PaymentsCaptureRequest> for Paymen
         ))
     }
 }
+
+impl PaymentCapture {
+    async fn create_capture_and_update_attempt(
+        state: &AppState,
+        authorized_payment_attempt: storage::PaymentAttempt,
+        storage_scheme: enums::MerchantStorageScheme,
+    ) -> CustomResult<(storage::Capture, storage::PaymentAttempt), errors::StorageError> {
+        state
+            .store
+            .insert_capture(
+                Self::make_capture(&authorized_payment_attempt),
+                authorized_payment_attempt,
+                storage_scheme,
+            )
+            .await
+    }
+    fn make_capture(authorized_attempt: &storage::PaymentAttempt) -> storage::CaptureNew {
```

Comment:
Instead of make_capture, create impl function for PaymentAttempt as make_new_capture. And Inside the function also use defaults to construct CaptureNew.